anyway/infographics_utils.py

```diff
@@ -643,60 +646,77 @@ def generate_items(self) -> None:
 
     @staticmethod
     def filter_and_group_injured_count_per_age_group(request_params: RequestParams):
-        import re
+        road_number = request_params.location_info["road1"]
 
-        data_of_ages = get_accidents_stats(
-            table_obj=InvolvedMarkerView,
-            filters=get_injured_filters(request_params.location_info),
-            group_by="age_group_hebrew",
-            count="age_group_hebrew",
-            start_time=request_params.start_time,
-            end_time=request_params.end_time,
+        query = (
+            db.session.query(InvolvedMarkerView)
+            .filter(InvolvedMarkerView.accident_timestamp >= request_params.start_time)
+            .filter(InvolvedMarkerView.accident_timestamp <= request_params.end_time)
+            .filter(
+                InvolvedMarkerView.provider_code.in_(
+                    [
+                        BE_CONST.CBS_ACCIDENT_TYPE_1_CODE,
+                        BE_CONST.CBS_ACCIDENT_TYPE_3_CODE,
+                    ]
+                )
+            )
+            .filter(
+                InvolvedMarkerView.injury_severity.in_(
+                    [
+                        BE_CONST.InjurySeverity.DEAD,
+                        BE_CONST.InjurySeverity.SEVERE,
+                    ]
+                )
+            )
+            .filter(
+                (InvolvedMarkerView.road1 == road_number)
+                | (InvolvedMarkerView.road2 == road_number)
+            )
+            .group_by("age_group_hebrew", "injury_severity")
+            .with_entities("age_group_hebrew", "injury_severity", func.count().label("count"))
         )
```

Comment:
Working with the code (i.e. clumn `age_group`) will be simpler and faster:
No need to convert text to numbers, 
No need to calculate the group in `dict_grouped` that a given row belongs to. Its first index can be the `age_group` code.
Instead of `range_dict`, a dict to translate the code to a pair of numbers can be used (located at `backend_constants.py` or `inforgraphics_disctionalries.py`).